libs/designsystem/menu/src/menu.component.ts

```diff
@@ -167,68 +171,75 @@ export class MenuComponent implements AfterViewInit, AfterContentInit, OnDestroy
           this.focusedIndex--;
         }
         this.focusItem();
-        this.preventFurtherPropagation(event);
+        this.preventDefaultAndStopImmediatePropagation(event);
         break;
       case 'Home': {
         if (this.focusedIndex > 0) {
           this.focusedIndex = 0;
           this.focusItem();
         }
-        this.preventFurtherPropagation(event);
+        this.preventDefaultAndStopImmediatePropagation(event);
         break;
       }
       case 'End': {
         if (this.focusedIndex < this.kirbyItems.length - 1) {
           this.focusedIndex = this.kirbyItems.length - 1;
           this.focusItem();
         }
-        this.preventFurtherPropagation(event);
+        this.preventDefaultAndStopImmediatePropagation(event);
         break;
       }
       case 'Escape':
         if (this.closeOnEscapeKey) {
           this.floatingMenu.hide();
         }
-        this.preventFurtherPropagation(event);
+        this.preventDefaultAndStopImmediatePropagation(event);
         break;
       case 'Tab':
         this.floatingMenu.hide();
         break;
       default: {
         if (this.isPrintableCharacter(key)) {
-          const foundItemIndex = this.getIndexByFirstCharacter(
-            key,
-            this.kirbyItems.map((item) => item.nativeElement.innerText)
-          );
+          const foundItemIndex = this.getIndexOfItemsByFirstCharacterFromItems(key);
           if (foundItemIndex > -1) {
             this.focusedIndex = foundItemIndex;
             this.focusItem();
           }
-          this.preventFurtherPropagation(event);
+          this.preventDefaultAndStopImmediatePropagation(event);
         }
       }
     }
   }
 
+  private getIndexOfItemsByFirstCharacterFromItems(char: string) {
+    return this.getIndexByFirstCharacter(
+      char,
+      this.kirbyItems.map((item) => item.nativeElement.innerText)
+    );
+  }
```

Comment:
Maybe not important, but could be simplified to a list of first char:
```suggestion
    return this.getIndexByFirstCharacter(
      char,
      this.kirbyItems.map((item) => item.nativeElement.innerText?.[0])
    );
  }
```

Reply:
Very nice optimization but i'd rather leave it as is for readability